instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/OpenTelemetryAutoConfiguration.java

```diff
@@ -65,17 +67,31 @@ public static class OpenTelemetrySdkConfig {
 
     @Bean
     @ConfigurationPropertiesBinding
-    @ConditionalOnBean({
-      OtelResourceAutoConfiguration.class,
-      OtlpLoggerExporterAutoConfiguration.class,
-      OtlpSpanExporterAutoConfiguration.class,
-      OtlpMetricExporterAutoConfiguration.class
-    })
+    @Conditional(MapConverterCondition.class)
     public MapConverter mapConverter() {
       // needed for otlp exporter headers and OtelResourceProperties
       return new MapConverter();
     }
 
+    static final class MapConverterCondition implements Condition {
+
+      private static final List<Class<?>> ANY_REQUIRED_BEANS =
+          Arrays.asList(
+              OtelResourceAutoConfiguration.class,
+              OtlpLoggerExporterAutoConfiguration.class,
+              OtlpSpanExporterAutoConfiguration.class,
+              OtlpMetricExporterAutoConfiguration.class);
+
+      @Override
+      public boolean matches(
+          org.springframework.context.annotation.ConditionContext context,
+          org.springframework.core.type.AnnotatedTypeMetadata metadata) {
+
+        return ANY_REQUIRED_BEANS.stream()
+            .anyMatch(s -> context.getBeanFactory().containsBean(s.getName()));
```

Comment:
while this checks for presence of a bean by name the javadoc in https://docs.spring.io/spring-boot/docs/current/api/org/springframework/boot/autoconfigure/condition/ConditionalOnBean.html indicates that it checks by type. Would it make sense to keep that behavior? Perhaps it would work if you'd use something like https://docs.spring.io/spring-boot/docs/current/api/org/springframework/boot/autoconfigure/condition/ConditionalOnBean.html where you put `@ConditionalOnBean` for each of these classes on a separate nested class?